### PR TITLE
Support more proposal states

### PR DESCRIPTION
**Pull request by alexkeating**
#### Description

- Update the contract deployments
- Support all the possible statuses

Closes https://github.com/ScopeLift/l2-flexible-voting-frontend/issues/148

### CODE REVIEW

**Comment by alexkeating** [comment]
Current dependencies on/for this PR:
* main
  * **PR #154** <a href="https://app.graphite.dev/github/pr/ScopeLift/l2-flexible-voting-frontend/154?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/ScopeLift/l2-flexible-voting-frontend/154?utm_source=stack-comment).

File: components/ProposalRow.tsx [hunk 1]
```diff
@@ -314,10 +314,19 @@ export default function ProposalRow({
...
 
 const ProposalState = ({ status }: { status: string }) => {
+  const green = ['bg-green-50', 'text-green-700', 'ring-green-600/20'];
+  const red = ['bg-red-50', 'text-red-700', 'ring-red-600/20'];
+  const gray = ['bg-gray-50', 'text-gray-700', 'ring-gray-600/20'];
   const colors: Record<string, string[]> = {
-    cancelled: ['bg-gray-50', 'text-gray-700', 'ring-gray-600/20'],
-    open: ['bg-green-50', 'text-green-700', 'ring-green-600/20'],
-    closed: ['bg-red-50', 'text-red-700', 'ring-red-600/20'],
+    suceeded: green,
```
Comment (wildmolasses) [inline components/ProposalRow.tsx:321]:
```suggestion
    succeeded: green,
```

File: components/ProposalRow.tsx [hunk 2]
```diff
@@ -314,10 +314,19 @@ export default function ProposalRow({
 }
 
 const ProposalState = ({ status }: { status: string }) => {
+  const green = ['bg-green-50', 'text-green-700', 'ring-green-600/20'];
```
Comment (wildmolasses) [inline components/ProposalRow.tsx:317]: love this pattern!

File: hooks/useProposals.ts [hunk 3]
```diff
@@ -104,7 +104,7 @@ const createFetcher =
     );
   };
 
-export const useL1Proposals = () => {
+export const useL1Proposals = ({ fetchProposalState }: { fetchProposalState?: boolean }) => {
```
Comment (wildmolasses) [inline hooks/useProposals.ts:107]: It seems we use this once and it's set to true, did I miss something? If we don't use, I'd prefer to cut the param.

File: hooks/useProposals.ts [hunk 4]
```diff
@@ -211,7 +229,7 @@ const useL2ProposalsState = (l2Proposals?: { proposalId: string; startBlock: str
     data: l2Proposals?.map((proposal, i) => {
       return {
         proposalId: proposal.proposalId,
-        state: proposalState?.[i].result,
+        state: proposalState?.[i].result === 6 ? 'closed' : proposalState?.[i].result,
```
Comment (wildmolasses) [inline hooks/useProposals.ts:232]: works for me; i might suggest a small comment here explaining why `6` needs to be special cased

File: hooks/useProposals.ts [hunk 5]
```diff
@@ -221,38 +239,44 @@ const useL2ProposalsState = (l2Proposals?: { proposalId: string; startBlock: str
...
   } else if (contractStatus === 1) {
-    return 'open';
+    return 'active';
   } else if (contractStatus === 2) {
-    return 'closed';
+    return 'cancelled';
+  } else if (contractStatus === 3) {
+    return 'defeated';
+  } else if (contractStatus === 4) {
+    return 'suceeded';
```
Comment (wildmolasses) [inline hooks/useProposals.ts:250]:
```suggestion
    return 'succeeded';
```

**Review by wildmolasses — APPROVED** [review]
looks good, a few small requests but good to merge after

**Comment by wildmolasses** [comment]
One more thing -- is it correct for L2 to show "closed" while L1 shows "pending"? If not, do we want to resolve here or in separate PR?
Screenshot 2023-11-28 at 12 20 31 PM

**Review by garyghayrat — APPROVED** [review]

File: hooks/useProposals.ts (hunk 3 above)
Comment (alexkeating) [reply hooks/useProposals.ts:107]: Yes, that is true. The idea is if we do reuse it we don't want the state fetch to always be bundled with the proposals. I can remove

File: components/ProposalRow.tsx (hunk 1 above)
Comment (alexkeating) [reply components/ProposalRow.tsx]: https://github.com/ScopeLift/l2-flexible-voting-frontend/pull/154/commits/2f2c309eb14c80b88ac5a6a31b3f57795612f6de

File: hooks/useProposals.ts (hunk 3 above)
Comment (alexkeating) [reply hooks/useProposals.ts]: https://github.com/ScopeLift/l2-flexible-voting-frontend/pull/154/commits/2f2c309eb14c80b88ac5a6a31b3f57795612f6de

File: hooks/useProposals.ts (hunk 4 above)
Comment (alexkeating) [reply hooks/useProposals.ts]: https://github.com/ScopeLift/l2-flexible-voting-frontend/pull/154/commits/2f2c309eb14c80b88ac5a6a31b3f57795612f6de

File: hooks/useProposals.ts (hunk 5 above)
Comment (alexkeating) [reply hooks/useProposals.ts]: https://github.com/ScopeLift/l2-flexible-voting-frontend/pull/154/commits/2f2c309eb14c80b88ac5a6a31b3f57795612f6de

**Comment by alexkeating** [comment]
> One more thing -- is it correct for L2 to show "closed" while L1 shows "pending"? If not, do we want to resolve here or in separate PR? Screenshot 2023-11-28 at 12 20 31 PM

Took a look, I think the proopsal has been fetched before the state has been fetched. Let's handle it in a followup pr.

**Pull request merged**